src/components/Insights.tsx

```diff
@@ -0,0 +1,112 @@
+import Navbar from './Navbar';
+import Router from "next/router";
+import { useRouter } from 'next/router';
+import { getCurrentLunarPhase } from '@/utils/lunar-phase';
+import Link from 'next/link';
+import CelestialLogo from './CelestialLogo';
+import { getEntry, getHoroscope } from '@/utils/apiCalls';
+import { Value } from 'react-calendar/dist/cjs/shared/types';
+import Image from 'next/image';
+import DatePicker from './DatePicker';
+import LoadingGif from './LoadingGif';
+
+type InsightsProps = AuthProps & {
+  updateEntryDate: (date: Value) => void;
+  selections: selectionType
+};
+
+export default function Insights({ isAuthorized, data, updateEntryDate, selections }: InsightsProps) {
+  const router = useRouter();
+  const { date } = router.query;
+  const [user, setUser] = useState<UserData | null>(null)
+  const [error, setError] = useState<boolean>(false)
+  const [emptyDay, setEmptyDay] = useState<boolean>(false)
+  const [userInsights, setUserInsights] = useState<Horoscope>()
+  const [chosenDate, setChosenDate] = useState<string>(date as string)
+  const [loading, setloading] = useState<boolean>(false)
+  
+  useEffect(() => {
+    if (!isAuthorized) {
+      Router.push("/");
+    }
+    setloading(true)
+
+    if (data) setUser(data[0])
+    if (user) getHoroscope(chosenDate, user?.zodiac_sign as string)
+    .then((data) => {
+      
+      if (data.length === 0) {
+        setEmptyDay(true)
+        setloading(false)
+      } else {
+        setUserInsights(data[0])
+        setloading(false)
+      }
+    })
+    .catch(err => {
+      setError(true)
+      console.error(err)
+    })
+    
+    return () => {
+      setloading(false)
+      setError(false)
+      setEmptyDay(false)
+    }
+  }, [isAuthorized, user, date]);
+
+  const goToEntry = () => {
+    updateEntryDate(convertStringToDate(chosenDate));
+    router.push(`${router.asPath.includes('demo') ? '/demo' : ''}/form`);
+  };
+
+  return (
+    <div className='relative h-full flex flex-col fade-in'>
+      <div className='mt-10 h-full'>
+        <CelestialLogo />
+        <DatePicker setChosenDate={setChosenDate} entryDate={convertStringToDate(date as string) as Date} updateEntryDate={updateEntryDate}/>
+        <h2 className='text-center celestial-cursive text-xl mt-10'>Today&#39;s Insights</h2>
+        <section className='insights mt-5 overflow-y-auto'>
+          <div className='flex justify-end mt-3 mr-5'>
+            <p className='text-lg'>{getCurrentLunarPhase(convertStringToDate(chosenDate) as Date).emoji} {getCurrentLunarPhase(convertStringToDate(chosenDate) as Date).description}</p>
+          </div>
+          <p className='p-5 insights-text text-lg'>
+            {error ? "Error loading insights, please refresh the page" : 
+              emptyDay ? "No insights loaded for this date, try a later date" : 
+              loading? <LoadingGif /> : userInsights?.description}
+          </p>
```

Comment:
I'm getting this error in the console: `Warning: validateDOMNesting(...): <p> cannot appear as a descendant of <p>.` It can be resolved by changing this <p> to a <div>